### PR TITLE
Updating to use oc replace and conditionally update edit and admin roles

### PR DESCRIPTION
**Pull request by ewolinetz**
Updating per comments on #4765 from @liggitt

### CODE REVIEW

File: roles/openshift_service_catalog/tasks/install.yml [hunk 1]
```diff
@@ -72,33 +72,49 @@
...
+# only do this if we don't already have the updated role info
 - name: Generate apply template for clusterrole/edit
   template:
     src: sc_role_patching.j2
     dest: "{{ mktemp.stdout }}/edit_sc_patch.yml"
   vars:
     original_content: "{{ edit_yaml.results.results[0] | to_yaml }}"
+  when:
+  - '"servicecatalog.k8s.io" not in edit_yaml.results.results[0]| to_yaml'
+  - '"settings.k8s.io" not in edit_yaml.results.results[0]| to_yaml'
```
Comment (liggitt) [inline roles/openshift_service_catalog/tasks/install.yml]: these checks seem pretty fragile, especially the quoting bits and the assumption that anything present from these groups means all intended permissions are present. we don't have a great tool for doing covers checks though...

Comment (ewolinetz) [inline roles/openshift_service_catalog/tasks/install.yml]: would there be a way we could do a `oc policy can-i --as` to confirm that these permissions are available? or would we need a user to be set up first; that might be a better way to verify that these permissions are available yet

**Comment by ewolinetz** [comment]
@liggitt created a filter plugin to evaluate if we have the specific rule contained already within the clusterrole

**Comment by ewolinetz** [comment]
Added commit to resolve https://bugzilla.redhat.com/show_bug.cgi?id=1471881

**Comment by ewolinetz** [comment]
@sdodson can you review the filter plugin and make sure this PR is still up to par?

File: roles/openshift_service_catalog/filter_plugins/contains_rule.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    '''Return true if the specified rule is contained within the provided source'''
+
+    rules=source['rules']
+
+    if rules:
+        for rule in rules:
+            if rule['apiGroups'].sort() == apiGroups.sort():
+                if rule['resources'].sort() == resources.sort():
+                    if rule['verbs'].sort() == verbs.sort():
+                        if rule['attributeRestrictions'] == attributeRestrictions:
```
Comment (liggitt) [inline roles/openshift_service_catalog/filter_plugins/contains_rule.py]: can drop attributeRestrictions, it's been deprecated and is no longer used

**Comment by liggitt** [comment]
thanks, detection + replace make me feel better about this

**Comment by sdodson** [comment]
@ewolinetz looks fine to me, but also looks like this isn't specific to service catalog, should we move this to the top level filter_plugins directory for general use?

**Comment by ewolinetz** [comment]
@sdodson will do

**Comment by sdodson** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
@sdodson  hold off on merging this, i think my `when` is incorrect... i dont see it performing the replace for the roles

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
resolved all travis issues, resolved issue was seeing where it wasn't correctly detecting that the roles weren't added yet

**Comment by sdodson** [comment]
@jlebon know what's happening in the fedora test? looks like a yum error on updates repo?

**Comment by jlebon** [comment]
@sdodson Looks like just regular flaky infra. You can always "bot, retest this please" if it happens.

**Pull request merged**